composables/use-collector-message.ts

```diff
@@ -0,0 +1,97 @@
+import { CollectorMessageModal } from '#components'
+import type { CollectorMessageModalProps } from '~/components/CollectorMessageModal.props'
+
+export function useCollectorMessage({
+  classId,
+  paymentId,
+  claimingToken,
+}: {
+  classId: string
+  paymentId: string
+  claimingToken: string
+}) {
+  const { user } = useUserSession()
+  const likeCoinSessionAPI = useLikeCoinSessionAPI()
+  const { t: $t } = useI18n()
+  const { handleError } = useErrorHandler()
+
+  const overlay = useOverlay()
+  const modal = overlay.create(CollectorMessageModal)
+
+  const isLoading = ref(false)
+  const hasSubmittedCollectorMessage = ref(false)
+  const bookInfoProps = ref<CollectorMessageModalProps>({
+    bookCoverSrc: '',
+    bookName: '',
+    bookAuthor: '',
+  })
+
+  function getModalProps() {
+    return {
+      ...bookInfoProps.value,
+      isLoading: isLoading.value,
+      hasSubmitted: hasSubmittedCollectorMessage.value,
```

Comment:
I think you can pass `isLoading` && `hasSubmitted` as ref to modal so that you dont need to update modal